pages/_middleware.ts

```diff
@@ -1,15 +1,33 @@
 import { NextMiddleware, NextRequest, NextResponse } from 'next/server';
+import { log } from '@/lib/log';
 
 // Next.js middleware
 // ==================
 // https://nextjs.org/docs/middleware
 
 export const middleware: NextMiddleware = ( req: NextRequest ) => {
+	const pathName = req.nextUrl.pathname;
+
 	// Required health check endpoint on VIP. Do not remove.
-	if ( req.nextUrl.pathname === '/cache-healthcheck' ) {
+	if ( pathName === '/cache-healthcheck' ) {
 		return new NextResponse( 'Ok' );
 	}
 
-	// Returning nothing means the request will continue as normal through the
-	// Next.js lifecycle.
+	// Ensures that when this response is returned, the Next.js chain will continue
+	const response = NextResponse.next();
+	const reqHeaderName = 'x-request-id';
+
+
+	if ( req.headers.has( reqHeaderName ) ) {
```

Comment:
Given our discussion, I don't really like that this runs conditionally, and not in local development because the request header isn't present. Have you thought more about the idea of minting a "local" request ID so that the behavior is consistent across environments?

I'm also still not sure about shipping this feature based on HTTP headers.  I think using `res.locals` might be a better approach now that we've done some validation of the idea.

Reply:
Pushed up a new change from our discussions that now does:

- Common method for building the requestContext
- Generate the id if it's not present

So it will rely on this requestcontext built from this method, that can be re-used outside the Apollo method as well rather than the header passing method.